ci/build_cpp.sh

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env bash
```

Comment:
if possible, this should be used in other Dockerfiles as well (`Dockerfile` and `Dockerfile-ubuntu`) to reduce duplication

Reply:
I do not understand your comment. Could you please expand?

Reply:
the CI builds are performed based on the two mentioned Dockerfiles, see for example https://github.com/DARMA-tasking/checkpoint/blob/f64650d0e680aa309facfcf2629f3101a1b8399a/Dockerfile#L25-L52

`build_cpp.sh` script could probably be used there to reduce duplication (although I do understand that this is not the goal of this task, so maybe this can be solved under a separate issue after your PR gets merged)

Reply:
Thank you for clarifying. 
If needed, I am willing to update the branch.
But I would agree that a separate issue may be more appropriate.